src/genegraph/transform/clinvar/common.clj

```diff
@@ -305,17 +297,125 @@ LIMIT 1")
 
 (defn fields-to-extension-maps
   "Returns a seq of Extension maps for each field in input-map.
-   If a value in input-map is a seq, create an Extension for each element."
+   If a value in input-map is a seq, and expand-seqs? is true,
+   create an Extension for each element."
+  ([input-map] (fields-to-extension-maps input-map {}))
+  ([input-map {:keys [expand-seqs?]}]
+   (->> (for [[k v] input-map]
+          (if (and expand-seqs? (sequential? v))
+            (->> (for [vi v]
+                   (fields-to-extension-maps {k vi}))
+                 (apply concat))
+            [{:type "Extension"
+              :name (name k)
+              :value v}]))
+        (apply concat))))
```

Comment:
This looks like a `reduce`.
Why define it here instead of variation.clj, where it is used?

Reply:
We will likely have other object types which will use this as well. The `clinical_assertion` transform module will probably have some extensions added in the near-ish future.

Reply:
When is `expand-seqs?` true.

Reply:
Is that just this? What am I missing?
```clojure
(defn fields-to-extension-maps
  "Return Extension maps for the key/value items from INPUT-MAP. "
  [input-map]
  (letfn [(extension [m [k v]]
            (conj m {:type "Extension" :name (name k) :value v}))]
    (reduce extension [] input-map)))
```

Reply:
@tbl3rd I've incorporated this feedback with the `expand-seqs?` option. This is a little easier to read than the `for` and `concat` implementation.
```clojure
(defn fields-to-extension-maps
  "Returns a seq of Extension maps for each field in input-map.
   If a value in input-map is a seq, and expand-seqs? is true,
   create an Extension for each element.
   example:
   (fields-to-extension-maps
    {:a :A :b :B :d [:D1 :D2]}
    {:expand-seqs? true})"
  ([input-map] (fields-to-extension-maps input-map {}))
  ([input-map {:keys [expand-seqs?]}]
   (letfn [(make-ext [agg [k v]]
             (if (and (sequential? v) expand-seqs?)
               (reduce make-ext agg (map #(vector k %) v)) ;; reduce pairs of k with each v
               (conj agg {:type "Extension" :name (name k) :value v})))]
     (reduce make-ext [] input-map))))
```